packages/components/datepicker/src/components/osds-datepicker/osds-datepicker.tsx

```diff
@@ -265,6 +265,7 @@ export class OsdsDatepicker implements OdsDatepickerAttribute, OdsDatepickerEven
           placeholder={placeholder}
           type={ODS_INPUT_TYPE.text}
           value={this.formatDate(value)}
+          style={{ '--placeholder-color': 'red' }}
```

Comment:
can't you use a class instead to keep css code outside of ts code?
replace 'red' with an actual value from the theme instead